.github/workflows/run_tests.yml

```diff
@@ -0,0 +1,127 @@
+name: Run Tests
+
+on:
+
+  pull_request:
+
+    branches:
+
+      - main
+
+jobs:
+
+  build:
+
+    strategy:
+
+      matrix:
```

Comment:
I don't think it's necessary to run datasci tests in all environments. Tools like polars, pandas and such should be cross-platform enough.